queries.sql

```diff
@@ -14,4 +14,52 @@ SELECT * FROM animals WHERE neutered = True;
 
 SELECT * FROM animals WHERE name != 'Gabumon';
 
-SELECT * FROM animals WHERE weight_kg >= 10.4 AND weight_kg <= 17.3;
\ No newline at end of file
+SELECT * FROM animals WHERE weight_kg >= 10.4 AND weight_kg <= 17.3;
+
+-- Begin Transaction
+
+BEGIN;
+
+UPDATE animals SET species = 'unspecified';
+
+ROLLBACK;
+
+-- Begin another Transaction
+
+BEGIN;
+
+UPDATE animals SET species = 'digimon' WHERE name LIKE '%mon';
+
+UPDATE animals SET species = 'pokemon' WHERE species IS NULL;
+
+COMMIT;
+
+-- Begin another Transaction
+
+BEGIN;
+
+DELETE FROM animals;
+
+ROLLBACK;
+
+-- Begin another Transaction 
+
+BEGIN;
+
+DELETE FROM animals WHERE date_of_birth > '2022-01-01';
+
+SAVEPOINT SP_1;
+
+UPDATE animals SET weight_kg = weight_kg * -1;
+
+ROLLBACK TO SP_1;
+
+UPDATE animals SET weight_kg = weight_kg * -1 WHERE weight_kg < 0;
+
+COMMIT;
+
+-- Answering questions with queries
+
+SELECT COUNT(*) FROM animals;
+
+SELECT COUNT(*) FROM animals WHERE escape_attempts = 0;
```

Comment:
You are missing four more queries highlighted in the project requirements below:
![image](https://user-images.githubusercontent.com/38088818/171429387-02114974-6a03-40e5-862c-050adfcfc96e.png)

- I will give you an example of the first two:

What is the average weight of animals?

```
   SELECT AVG(weight_kg) FROM animals;
```
Who escapes the most, neutered or not neutered animals?

```
   SELECT neutered, AVG(escape_attempts) FROM animals
   GROUP BY neutered;
```

- Kindly finish the rest of the queries by writing the SQL for `What is the minimum and maximum weight of each type of animal?` and `What is the average number of escape attempts per animal type of those born between 1990 and 2000?`.